command/validate.go

```diff
@@ -42,6 +44,11 @@ func (c *ValidateCommand) Run(args []string) int {
 		return 1
 	}
 
+	// If set, output a warning indicating that these values are not used.
+	if !varValues.Empty() || !varFiles.Empty() {
+		c.Ui.Warn("The -var and -var-file flags are not used in validate. Setting them has no effect.")
```

Comment:
This message text looks good to me but `c.Ui.Warn` is the "old" way to produce warnings, which produces big blocks of yellow text that are hard to read on some terminals.

Might be worth moving the `var diags tfdiags.Diagnostics` line below up here and appending the warning to the diagnostics instead. As well as being cosmetically nicer due to using our new diagnostics printing, in this particular command I think that will also mean the warning gets output in JSON format when running in JSON mode, which isn't strictly necessary but less likely to cause trouble for existing software that is consuming the JSON validate output that might not be expecting to find content on `stderr` in that case.